src/main/scala/codacy/pylint/Pylint.scala

```diff
@@ -1,73 +1,85 @@
 package codacy.pylint
 
+import java.io.File
 import java.nio.file.{Files, Path}
 
-import codacy.dockerApi._
-import codacy.dockerApi.utils.{CommandRunner, FileHelper, ToolHelper}
+import codacy.docker.api._
+import codacy.docker.api.utils.ToolHelper
+import codacy.dockerApi.utils.{CommandRunner, FileHelper}
 import play.api.libs.json._
 
 import scala.sys.process._
 import scala.util.{Properties, Success, Try}
 
 object Pylint extends Tool {
 
-  override def apply(path: Path, conf: Option[List[PatternDef]], files: Option[Set[Path]])(implicit spec: Spec): Try[List[Result]] = {
-    val completeConf = ToolHelper.getPatternsToLint(conf)
+  private val pythonVersionKey = Configuration.Key("python_version")
+  private val python3 = "3"
+
+  def apply(source: Source.Directory, configuration: Option[List[Pattern.Definition]], files: Option[Set[Source.File]],
+            options: Map[Configuration.Key, Configuration.Value])
+           (implicit specification: Tool.Specification): Try[List[Result]] = {
+    val completeConf = ToolHelper.patternsToLint(configuration)
 
     def isEnabled(issue: Result) = {
       issue match {
-        case Issue(_, _, patternId, _) => completeConf.map(item => item.exists(_.patternId == patternId)).getOrElse(true)
+        case Result.Issue(_, _, patternId, _) => completeConf.forall(item => item.exists(_.patternId == patternId))
         case _ => true
       }
     }
 
     def buildFileCommands(files: Map[String, Array[String]]) = {
-      files.map { case (key, values) => commandFor(key, path, completeConf, values) }
+      files.map { case (key, values) => commandFor(key, completeConf, values) }
         .flatMap(item => item.toOption).toList
     }
 
     def getStdout(command: List[String]): Try[List[String]] = {
       Try {
-        CommandRunner.exec(command, Some(path.toFile)) match {
+        CommandRunner.exec(command, Option(new File(source.path))) match {
           case Right(resultFromTool) =>
             resultFromTool.stdout
-          case Left(failure) => {
+          case Left(failure) =>
             throw failure
-          }
         }
       }
     }
 
-    val collectedFiles = collectFiles(files, path)
-    val classified = classifyFiles(collectedFiles)
-    val commands = classified.map { case item => buildFileCommands(item) }
+    val collectedFiles = collectFiles(files, source)
+    val classified = options.get(pythonVersionKey).fold {
+      classifyFiles(collectedFiles)
+    } { pythonVersion =>
+      val validPythonVersion = Option(pythonVersion).collect {
+        case JsNumber(version) => s"$version"
```

Comment:
I think we should accept string here too. `"3"` should be valid too. Json will not parse this as a number.